clash-prelude/src/Clash/Signal/Delayed/Bundle.hs

```diff
@@ -219,3 +220,20 @@ instance KnownNat d => Bundle (RTree d a) where
   type Unbundled t delay (RTree d a) = RTree d (DSignal t delay a)
   bundle   = sequenceA
   unbundle = sequenceA . fmap lazyT
+
+-- | Internal data type, see "TaggedUnit".
+data TaggedUnit' (d :: k) (dom :: i) = TaggedUnit
+
+-- | Same as "Clash.Signal.Bundle", but adapted for "DSignal".
+type TaggedUnit (d :: Nat) (dom :: Domain) = TaggedUnit' '(d, d) '(dom, dom)
```

Comment:
Does this need to be `(d :: Nat)`, or can it be `(d :: k)` too?

Reply:
Could, but I thought this would be a little bit more self-documenting